weave

```diff
@@ -641,16 +653,20 @@ case "$COMMAND" in
             DNS_ARGS=$(exec_remote dns-args "$@")
             [ "$1" = "--with-dns" ] && shift 1
         fi
-        CIDR=$1
-        shift 1
-        validate_cidr $CIDR
+        CIDRS=""
+        while is_cidr "$1"; do
+            CIDRS="$CIDRS $1"
+            shift 1
+        done
```

Comment:
This `while` over `is_cidr` appears in four places, but I'm struggling to see how to factor it out as it modifies `$@` with `shift`. Any thoughts @dpw?